### PR TITLE
Docs for Button Component & removed redundant condition in <Button />

### PR DESCRIPTION
**Pull request by deadcoder0904**
So I just created wrote Button docs, please check it & let me know if it needs improvement & I also removed a redundant condition in the Button Component. I will be completing other tasks till then 😉

### CODE REVIEW

File: docs/API/buttons.md [hunk 1]
```diff
@@ -1,68 +1,68 @@
-Buttons take a title and an optional [material icon name](https://design.google.com/icons/), as well as the props below.
-
-> You can override Material icons with one of the following: material-community, simple-line-icon, zocial, font-awesome, octicon, ionicon, foundation, evilicon, or entypo by providing an icon.type as a prop.
+Buttons can be used to interact with the screen. It takes the following props.
 
 ![Buttons](http://i.imgur.com/CVf1xbr.png)
```
Comment (Monte9) [inline docs/API/buttons.md]: we will need to update this screenshot with the new buttons layout. You could just take a screenshot from the example expo app.

File: docs/API/buttons.md [hunk 2]
```diff
@@ -1,68 +1,68 @@
...
+| clear | none | boolean | makes the button transparent (optional) |
+| loading | none | boolean | prop to display a loading spinner (optional) |
+| loadingStyle | none | View style (object) | add additional styling for loading component (optional) |
+| loadingProps | none | object (style) | add additional props for ActivityIndicator component (optional) |
+| onPress | none | function | onPress method (optional) |
+| containerStyle | none | View style (object) | styling for Component container |
+| icon | none | React Native Component | displays a centered icon (when no text) or to the left (with text). (can be used along with rightIcon as well) |
+| iconContainerStyle | none | View style (object) | styling for Component container |
+| iconRight | none | boolean | displays a centered icon (when no text) or to the left (with text). (can be used along with rightIcon as well) |
+| linearGradientProps | none | object | displays a linear gradient |
```
Comment (Monte9) [inline docs/API/buttons.md]: wow! This clean up is 💯. Good job! 👏

Comment (Monte9) [inline docs/API/buttons.md]: also for `linearGradientProps` let's change the description to `displays a linear gradient (supports Expo only)`

File: docs/API/buttons.md [hunk 3]
```diff
@@ -1,68 +1,68 @@
...
 
-> Also recevies all TouchableWithoutFeedback props
+> Also receives all TouchableNativeFeedback (Android) or TouchableOpacity (iOS) props
 
 | prop | default | type | description |
 | ---- | ---- | ----| ---- |
-| containerViewStyle | none | View style (object) | styling for Component container |
-| Component | TouchableHighlight (iOS), TouchableNativeFeedback (android) | React Native Component | Specify other component such as TouchableOpacity or other (optional) |
+| text | none | string | button text (optional) |
+| textStyle | none | Text style (object) | add additional styling for text component (optional) |
```
Comment (Monte9) [inline docs/API/buttons.md]: let's capitalize the T in `Text Component`

File: docs/API/buttons.md [hunk 4]
```diff
@@ -1,68 +1,68 @@
...
-| rounded | false | boolean | rounds the button (optional) |
-| outline | false | boolean | outlines the button (optional) |
-| transparent | false | boolean | makes the button transparent (optional) |
+| clear | none | boolean | makes the button transparent (optional) |
+| loading | none | boolean | prop to display a loading spinner (optional) |
+| loadingStyle | none | View style (object) | add additional styling for loading component (optional) |
+| loadingProps | none | object (style) | add additional props for ActivityIndicator component (optional) |
+| onPress | none | function | onPress method (optional) |
+| containerStyle | none | View style (object) | styling for Component container |
+| icon | none | React Native Component | displays a centered icon (when no text) or to the left (with text). (can be used along with rightIcon as well) |
```
Comment (Monte9) [inline docs/API/buttons.md]: we don't use `rightIcon` anymore. Please replace it with `iconRight`

File: docs/API/buttons.md [hunk 5]
```diff
@@ -1,68 +1,68 @@
...
+| iconContainerStyle | none | View style (object) | styling for Component container |
```
Comment (Monte9) [inline docs/API/buttons.md]: change description to `styling for Icon Component container`

File: docs/API/buttons.md [hunk 6]
```diff
@@ -1,68 +1,68 @@
...
+| iconRight | none | boolean | displays a centered icon (when no text) or to the left (with text). (can be used along with rightIcon as well) |
```
Comment (Monte9) [inline docs/API/buttons.md]: change the description to `displays Icon to the right of Text. Needs to be used along with icon prop`

File: src/buttons/Button.js [hunk 7]
```diff
@@ -69,30 +65,34 @@ class Button extends Component {
...
-              iconRight &&
-              <View style={[styles.iconContainer, iconContainerStyle]}>
-                {icon}
-              </View>}
+              iconRight && (
+                <View style={[styles.iconContainer, iconContainerStyle]}>
+                  {icon}
+                </View>
+              )}
           </ButtonContainer>
```
Comment (Monte9) [inline src/buttons/Button.js]: these changes look good.

**Comment by deadcoder0904** [comment]
I'll make the changes now

File: docs/API/buttons.md (hunk 1 above)
Comment (deadcoder0904) [reply docs/API/buttons.md]: which buttons layout ?

**Comment by Monte9** [comment]
Great thanks @deadcoder0904! Also please update the readme with the todo item that you just finished. https://github.com/react-native-training/react-native-elements#v10-todo

So just check the appropriate box & put your name to the right of it so we can keep track of all our contributors. 

As far as the payout is concerned, we do that through open collective. So you would have to sign up on their website [here](https://opencollective.com/) and then submit an expense to RNE [here](https://opencollective.com/react-native-elements/expenses/new). 

Once you submit a expense I can approve it and they will send it to your paypal within a week. Let me know if you have any other q's and thanks again for the PR 🎉

**Comment by deadcoder0904** [comment]
Ok so should I complete all or should I create an issue that stating I'm completing it❓

File: README.md [hunk 8]
```diff
@@ -113,7 +113,7 @@ Current release: `1.0.0-beta`
 
 ### v1.0 TODO:
 
-- [ ] Update docs for Button component - **[$20](https://opencollective.com/react-native-elements#budget)**
+- [x] Update docs for Button component - **[$20](https://opencollective.com/react-native-elements#budget)** - @deadcoder0904
```
Comment (Monte9) [inline README.md]: let's get rid of the $20 too. Just don't forget to submit a expense on Open Collective

File: docs/API/buttons.md (hunk 1 above)
Comment (Monte9) [reply docs/API/buttons.md]: The one on the React Native Elements App. Have seen that yet? Here is the repo: https://github.com/react-native-training/react-native-elements-app and here is the expo app you can run on your phone: https://expo.io/@monte9/react-native-elements-app

if you run that locally, you will see that it has a buttons tab with all the types of buttons displayed on it. I suggest you screenshot that put together something presentable.

Comment (Monte9) [reply docs/API/buttons.md]: If this sounds like too much work, don't worry about it. I'll follow up on it.

File: docs/API/buttons.md [hunk 9]
```diff
@@ -1,68 +1,69 @@
...
-  iconRight={{name: 'code'}}
-  title='LARGE WITH RIGHT ICON' />
+  icon={
+    <Icon
+      name='arrow-right'
+      size={15}
+      color='white'
+    />
+  }
+  text='BUTTON WITH ICON' />
```
Comment (Monte9) [inline docs/API/buttons.md]: let's format all these button with `/>` on the next line

File: docs/API/buttons.md [hunk 10]
```diff
@@ -1,68 +1,69 @@
...
-  title='LARGE WITH ICON TYPE' />
+  icon={
+    <Icon
+      name='arrow-right'
+      size={15}
+      color='white'
+    />
+  }
+  iconRight
+  text='BUTTON WITH RIGHT ICON' />
```
Comment (Monte9) [inline docs/API/buttons.md]: same formatting comment here

File: docs/API/buttons.md [hunk 11]
```diff
@@ -1,68 +1,69 @@
...
+  textStyle={{ fontWeight: "700" }}
+  buttonStyle={{
+    backgroundColor: "rgba(92, 99,216, 1)",
+    width: 300,
+    height: 45,
+    borderColor: "transparent",
+    borderWidth: 0,
+    borderRadius: 5
+  }}
+  containerStyle={{ marginTop: 20 }} />
```
Comment (Monte9) [inline docs/API/buttons.md]: and here too

File: docs/API/input.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+| color | black | string | color of icon (optional) |
+| iconStyle | inherited style | object (style) | additional styling to icon (optional) |
+| component | View if no onPress method is defined, TouchableHighlight if onPress method is defined | React Native component | update React Native Component (optional) |
+| onPress | none | function | onPress method for button (optional) |
+| onLongPress | none | function | onLongPress method for button (optional) |
+| underlayColor | icon color | string | underlayColor for press event |
+| reverse | false | boolean | reverses color scheme (optional) |
+| raised | false | boolean | adds box shadow to button (optional) |
+| containerStyle | inherited styling | object (style) | add styling to container holding icon (optional) |
+| reverseColor | white | string | specify reverse icon color (optional) |
```
Comment (Monte9) [inline docs/API/input.md]: @deadcoder0904 can we do this in another PR? That way we don't add mix up multiple things in 1 PR. 

So please delete this file & create another PR for the input docs. thanks

**Comment by Monte9** [comment]
@deadcoder0904 for now, only the `<Input />` docs can be done. So if you can do that, please create another PR for it.

The rest of the TODO's are about docs for components that are in progress such as docs for search & web & themes etc. So will have to come back to these docs once the main features are merged into v1.

**Comment by deadcoder0904** [comment]
Ohh I mistakenly added `<Input />` by doing `git add .`

File: docs/API/buttons.md (hunk 1 above)
Comment (deadcoder0904) [reply docs/API/buttons.md]: Added this
